AngularApp/src/app/diagnostic-data/components/time-series-graph/time-series-graph.component.ts

```diff
@@ -134,8 +151,45 @@ export class TimeSeriesGraphComponent extends DataRenderBaseComponent implements
     });
   }
 
+  private _getGreatestCommonFactor(timestamp: moment.Moment) {
+    return this._gcd(timestamp.minutes(), 60)
+  }
+
+  private _gcd(a: number, b: number) {
+    a = Math.abs(a);
+    b = Math.abs(b);
+    if (b > a) {var temp = a; a = b; b = temp;}
+    while (true) {
+        if (b == 0) return a;
+        a %= b;
+        if (a == 0) return b;
+        b %= a;
+    }
+  }
+
+  private _getDefaultTimegrain() {
+    let range = Math.abs(this.startTime.diff(this.endTime, 'hours'));
+    // > 7 days: 1 day
+    if (range > 168) {
+      return 24 * 60;
+    }
+    // else 1 hr
+    return 60;
```

Comment:
if the time range is less than 7 days, the default grain is 1 hour.
Should we include  :  
time range > 1 day ==> 1 hour time grain
time range <= 1 day ==> 5 min time grain

Reply:
The way that this works, if I make it 5 minutes for <= day, it would mean that you could never have a larger time grain than that. I agree that the ones you proposed make sense when there is no data, but if someone wanted to group by hour for a day period, I wanted to allow that. But if you think it is fine to enforce 5 minutes, then I can do that as well.